download/download_test.go

```diff
@@ -0,0 +1,336 @@
+package download
```

Comment:
Can I recommend you look at the testing of the Files client https://github.com/ONSdigital/dp-api-clients-go/blob/main/files/client_test.go

We took a different approach to testing which enabled us to simplify the client interface buy removing a number of functions that only exist for testing like URL() and HealthClient()